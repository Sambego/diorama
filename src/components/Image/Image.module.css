.container {
  position: relative;
}

.image {
  width: 100%;
  max-width: 80vw;
  max-height: 80vh;
  object-fit: cover;
  object-position: center;
}

.full {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  width: 100%;
  height: 100%;
  max-width: 100%;
  max-height: 100%;
}

.overlay {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  width: 100%;
  height: 100%;
  mix-blend-mode: multiply;
}

.contain {
  object-fit: contain;
}
