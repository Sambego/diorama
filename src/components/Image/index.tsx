import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";
import styles from "./Image.module.css";

type ImageProps = React.PropsWithChildren<{
	className?: string;
	style?: React.CSSProperties;
	color?: React.CSSProperties["backgroundColor"];
	full?: boolean;
	contain?: boolean;
	src: string;
	alt: string;
}>;

function Image({
	alt,
	src,
	style,
	full,
	color,
	className,
	contain,
}: ImageProps) {
	const containerClasses = classnames(
		styles.container,
		{
			[styles.full]: full,
		},
		"diorama-image-container"
	);
	const imageClasses = classnames(
		styles.image,
		{
			[styles.full]: full,
			[styles.contain]: contain,
			...(className ? { [className]: className !== "" } : {}),
		},
		"diorama-image"
	);

	return (
		<div className={containerClasses}>
			{color && (
				<div
					className={`${styles.overlay} diorama-image-overlay`}
					style={{ backgroundColor: color }}
				/>
			)}
			<img className={imageClasses} src={src} style={style} alt={alt} />
		</div>
	);
}

Image.propTypes = {
	alt: PropTypes.string.isRequired,
	color: PropTypes.string,
	className: PropTypes.string,
	contain: PropTypes.bool,
	full: PropTypes.bool,
	style: PropTypes.shape({}),
	src: PropTypes.string.isRequired,
};

Image.defaultProps = {
	color: undefined,
	className: "",
	contain: false,
	full: false,
	style: {},
};

export default Image;
