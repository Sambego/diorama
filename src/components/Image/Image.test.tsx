import React from 'react';
import renderer from 'react-test-renderer';
import Image from ".";

import test from '../../../__mocks__/test.png';

it('renders correctly', () => {
  const tree = renderer.create(<Image src={test} alt="image description" />).toJSON();
  expect(tree).toMatchSnapshot();
});
