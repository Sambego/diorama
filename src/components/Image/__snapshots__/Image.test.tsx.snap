// Vitest Snapshot v1

exports[`renders correctly 1`] = `
<div
  className="_container_1nb4n_1 diorama-image-container"
>
  <img
    alt="image description"
    className="_image_1nb4n_5 diorama-image"
    src="/__mocks__/test.png"
    style={{}}
  />
</div>
`;
